gensim/similarities/docsim.py

```diff
@@ -186,11 +186,27 @@ def __init__(self, output_prefix, corpus, num_features, num_best=None, chunksize
         `self.num_best = 10` before doing a query.
 
         """
+        self.is_shard = False  # To indicate if shard directory has been created, if not then shard and index files are stored in temp
         if output_prefix is None:
             # undocumented feature: set output_prefix=None to create the server in temp
             self.output_prefix = utils.randfname(prefix='simserver')
         else:
-            self.output_prefix = output_prefix
+            try:
+                self.output_prefix = os.path.join(os.path.dirname(output_prefix), 'shard', '')   # '' to get the trailing slash
+                if not os.path.exists(self.output_prefix):
+                    os.makedirs(self.output_prefix)
+                    logger.info("created %s directory . Both shard and index files are stored at this location. Please use \
+                    this entire folder for portability across different systems. ", self.output_prefix)
+                    self.is_shard = True  # helps while saving the index file as we don't want to create directories in temp
+                else:
+                    logger.warning("It seems there is already a file named shard in location %s . Please ensure  \
+                    that there is no other files named shard in the provided location. Starting similarity index under temp folder", os.path.join(os.path.dirname(output_prefix), ''))
+                    self.output_prefix = utils.randfname(prefix='simserver')
+            except:
+                logger.warning("Failed to create shard folder in %s. Please ensure that you have required permission to create  \
```

Comment:
yes, we discussed similar thing on hangouts with @tmylk . We don't need to create a directory.

My implementation was because of one observation - that if input is something like `\home\prakhar\` then we will be creating `.0` , `.1` which will be a hidden file, little difficult for user to locate and move (especially when we have different `name` for shards and pickle). Therefore, I tried for folder approach. (but that would be bad API, I agree)

I think, it's a better approach to ask user for only one input `prefix` and not again while saving. We can raise error if user tries to input a location which is a sub-directory only.
For example-

```python
dir, file_name = os.path.split(input_prefix)
# raise error if file_name is empty
```